### PR TITLE
Feature: control RGB LED

### PR DESCRIPTION
**Pull request by Olavhaasie**
Closes PM-285

## Description
Added listening to some topics in order to set the RGB led to some predefined colors.

**NOTE:** I still have to test it, but I do not have an Arduino right now, so I figured you could take a look already :wink: 

## Changes
* Added `Color` struct
* Updated `ros_lib`
* Added listening for error, instruction response and current gait

### CODE REVIEW

File: lib/ros_lib/march_shared_resources/GaitInstruction.h [hunk 1]
```diff
@@ -77,7 +78,7 @@ namespace march_shared_resources
     }
 
     const char * getType(){ return "march_shared_resources/GaitInstruction"; };
-    const char * getMD5(){ return "f655c02301e6cc940112aec0256d56ae"; };
+    const char * getMD5(){ return "c548bd1201c18293dcff8514a06fc253"; };
```
Comment (Joris-JW) [inline lib/ros_lib/march_shared_resources/GaitInstruction.h:81]: Got a question; what does this string represent?

Comment (Joris-JW) [reply lib/ros_lib/march_shared_resources/GaitInstruction.h:81]: gonna answer my own question here; it is automatically generated....

**Review by RutgerVanBeek — COMMENTED** [review]
Wouldn't it be easier to adjust the led if you move the callbacks for the topics to the main software and that this just listens to one topic to where colors are published? Adding a new light would be as simple as publishing on that topic.

**Comment by Olavhaasie** [comment]
> Wouldn't it be easier to adjust the led if you move the callbacks for the topics to the main software and that this just listens to one topic to where colors are published? Adding a new light would be as simple as publishing on that topic.

I thought the exact opposite. I think it is nicer if the rest of the software does not depend on knowing there exists one RGB led. If we want we can implement other types of feedback relatively easy now, since we only have to add the hardware implementation here. We could improve the interface on topics, since we have to listen to two different topics in order to know what gait is being executed.

**Review by RutgerVanBeek — DISMISSED** [review]
Okay, that is also a possibility. I don't really have comments, but also no experience with Arduino programming.

**Review by Joris-JW — DISMISSED** [review]

**Comment by Olavhaasie** [comment]
Had a bug where the color would always be set to the IDLE color when nothing was active. I changed it to set the color to IDLE only when it connects and when a gait finishes. Also, the red and green pins had to be swapped :shrug:

**Review by RutgerVanBeek — APPROVED** [review]
Would be nice to find some way to keep the LED active when the connection with ROS is lost.

**Pull request merged**